site/docs/BestPractice-ThinkingInGraphs.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Thinking in Graphs
+layout: ../_core/DocsLayout
+category: Best Practices
+permalink: /learn/thinking-in-graphs/
+---
+
+### It's Graphs All the Way Down
+> With GraphQL, you model your business domain as a graph
+
+Graphs are powerful tools for modeling many real-world phenomena because they resemble our natural mental models and verbal descriptions of the underlying process. With GraphQL, you model your business domain as a graph by defining a schema; within your schema, you define different types of nodes and how they connect/relate to one another. To the client, this pattern exposes a graph of entities and relationships that describe the business domain without implementation details.
+
+Before defining your schema, examine the everyday language you use to describe your business domain rules and processes. For example, let's try to describe an email app in plain english:
+
+* A user can have multiple email accounts
+* Each email account has an address, inbox, drafts, deleted items, and sent items
+* Each email has a sender, receive date, subject, and body
+* Senders have an email address
+* Users cannot send an email without a recipient address
+
+Naming things is a hard but important part of building intuitive APIs, so take time to carefully think about what makes sense for your problem domain and users. Your team should develop a shared understanding and consensus of these business domain rules because you will need to choose intuitive, durable names for nodes and relationships in the GraphQL schema. Try to imagine some of the queries you will want to execute:
```

Comment:
I think in most cases, programmers wouldn't think of these as "nodes in a graph" - they would think of them as objects in object-oriented programming, or entries in a database. Do you want to mention something around that? In particular, I have met a lot of people who think graphql is for graph databases. It isn't - it's really nice for both regular databases, and regular object-oriented programming. Those are the two concepts that I think most people will be mapping GraphQL to, here.